rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -286,6 +286,10 @@ private int tryDoEnd(boolean compensate) {
             int twoPhaseOutcome = retryGetEndStatus(endPath, compensate);
 
             if (twoPhaseOutcome != -1) {
+                if (parentId != null && status == ParticipantStatus.Completed) {
```

Comment:
will the execution ever get here? Because this method will be called again through recovery with compensate == false and the LRARecord is still completed so why do you not finish on line 271 again?

Reply:
Yes it can if the participant has Completed or Compensated or if a local transaction has already Closed or Cancelled.